walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,126 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    ArrayList<String> attendees = new ArrayList<String>(request.getAttendees());
+    long longDuration = request.getDuration();
+    ArrayList<TimeRange> possibleTimes = new ArrayList<TimeRange>();
+
+    // Deal with special cases:
+    // Meeting is too long, impossible to ever hold it
+    if (longDuration > TimeRange.WHOLE_DAY.duration()) {
+      return possibleTimes;
+    }
+    // No events/no attendees, hold meeting anytime
+    if (events.size() == 0 || attendees.size() == 0) {
+      possibleTimes.add(TimeRange.fromStartDuration(TimeRange.START_OF_DAY, TimeRange.WHOLE_DAY.duration()));
+      return possibleTimes;
+    }
+ 
+    // Since we know longDuration is less than one day, it can fit in an int
```

Comment:
Because later in the file, I want to check if I can fit duration amount of time between two meetings. But meetings' start and end times are ints, so they can't compare to longs.